service/src/main/java/org/jboss/aerogear/unifiedpush/service/impl/AliasServiceImpl.java

```diff
@@ -257,6 +269,33 @@ public void createAsynchronous(Alias alias) {
 		create(alias);
 	}
 
+	@Override
+	public int updateKCUsersGuids() {
+		Map<String, Set<UserTenantInfo>> aliasToIdentifiers = aliasDao.findAllUserTenantRelations()
+				.collect(Collectors.groupingBy(row -> {
+							String alias = row.getString(1);
+							return alias.toLowerCase(); // alias is case insensitive
+						},
+						Collectors.mapping(row -> {
```

Comment:
The code for reading the Row should best be encapsulated in the Dao Tier and not leak in to the Service Tier. 
see for example  `org.jboss.aerogear.unifiedpush.cassandra.dao.impl.NoSQLUserDaoImpl#getKey` as used in `org.jboss.aerogear.unifiedpush.cassandra.dao.impl.NoSQLUserDaoImpl#remove(java.util.UUID, java.lang.String)`